addon/mixins/data-adapter-mixin.js

```diff
@@ -56,6 +57,39 @@ export default Ember.Mixin.create({
   authorizer: null,
 
   /**
+    First authorizes the request and then calls base class' `ajax()`.
+
+    @param {String} url The requested URL.
+    @param {String} type The method of the HTTP request.
+    @param {Object} options Additional XHR options.
+  */
+  ajax(url, type, options) {
+    // tests benefit from doing it like this (instead of just `options = options || {}`).
+    if (!options) {
+      options = {};
+    }
+    const authorizer = this.get('authorizer');
+    Ember.assert("You're using the DataAdapterMixin without specifying an authorizer. Please add `authorizer: 'authorizer:application'` to your adapter.", Ember.isPresent(authorizer));
+
```

Comment:
The below is a workaround to be able to call `this._super()` in the promise handler. I spent quite a bit of time on this. Googling tells me that others are having issues with mixins, promises and calling `_super()` too. I tried various tricks (like passing `this` explicitly`) but this was the simplest workaround I could come up with.

If anybody know a better way to overcome this, I'd be glad to hear it.